### ISSUE TITLE
did we make a mistake with our conversion from physical quantity format to multiple fields?

### ISSUE THREAD

**Issue by schristley** [airr-community/airr-standards#504]
After thinking about @franasa recent issues, I wonder if we made a mistake. That is, before with the `physical quantity` format, the data was in a single field, but now it's spread across multiple fields and we've "lost" (or are lacking) any association between those fields. For example, programs have to "know" that if they are working with `collection_time_point_relative`, that it has to also work with `collection_time_point_relative_unit`, but there is nothing in the schema that indicates those fields are tied together.

This started with `age` when we split it out into 3 fields, and maybe that seemed okay at the time because it was a one-off, but we have a few of these physical quantity fields.

Should we instead have kept the fields "grouped" together into an object, like so:

```
age: { min: 10, max: 10, unit: { id: UO:0000036, label: year } }
collection_time_point_relative: { quantity: 12, unit: { id: UO:0000033, label: day } }
template_amount: { quantity: 12, unit: { id: UO:0000024, label: nanogram } }
```

It still then is "one field" from a MiAIRR perspective, and you can see for example that `template_amount` isn't being documented properly in the [data table in the docs](https://docs.airr-community.org/en/stable/miairr/data_elements.html).

**Comment by bussec** [comment]
In general, this makes sense to me, as I prefer records (read: nesting) in contrast to suffixed keys. Having say this, I would like to check whether this would be compatible with a description of the current fields using EFO (experimental facts ontology).

Furthermore, I think we should revisit the decision to describe the temporal order of events (sampling, enrollment, diagnosis) using a relative framework. We had privacy in mind when we did this, but there are potentially other ways to do it that are less error prone (e.g., noisy dates). But that's probably something for another issue.

**Comment by bcorrie** [comment]
I think we are trying to capture entities that have units as a single object.

To be consistent we may want to do:

```
age_min: { quantity: 10, unit: { id: UO:0000036, label: year } }
age_max: { quantity: 10, unit: { id: UO:0000036, label: year } }
```

This is represented as two entities with quantity and units. Embedding a object that encapsulates a range object seems confusing. You then need to know that there are two quantities with the same unit...

If you wanted to group them then:
```
age_range: { min: { quantity: 10, unit: { id: UO:0000036, label: year }, max: { quantity: 10, unit: { id: UO:0000036, label: year }}}
```

**Comment by schristley** [comment]
`age` is our special case where it can be either a single time point or a range, and we came up with special rules for it. If we do decide to go with nesting, maybe we also rethink the design to be more explicit.

Say, use this when it's a time point

```
age: { point: { quantity: 10, unit: { id: UO:0000036, label: year } } }
```

and use this for a time range

```
age: { range: { min: { quantity: 10, unit: { id: UO:0000036, label: year }, max: { quantity: 10, unit: { id: UO:0000036, label: year }}}}
```

the existent of `point` or `range` in the object tells you what it is.

**Comment by schristley** [comment]
@bcorrie What's your opinion for `age`? I tend to like stuffing everything in a single `age` field like [this](https://github.com/airr-community/airr-standards/issues/504#issuecomment-802158376), but what about internal structure of age? Keep our special rules with `min` and `max` to handle both points and ranges, or make them separate?

**Comment by bcorrie** [comment]
I think my main concern would be around representation/storage and query/analysis. It feels like this makes the decisions about storing data and analysis more complex. For example:

- If I have a point can I store it as a range with min and max the same (as we do now).
- If I am searching for an age point (e.g. 42), what does it mean when I perform that search on both point and range data?  

When everything is a range, you have only one way of doing these things and your query is explicit. You have to specify which field you are interested in (min and/or max) and the operator used on each field. If as a consumer of the data, if I have to consider what I mean with age ranges and age points things seem more complicated.

**Comment by schristley** [comment]
> When everything is a range, you have only one way of doing these things and your query is explicit. You have to specify which field you are interested in (min and/or max) and the operator used on each field. If as a consumer of the data, if I have to consider what I mean with age ranges and age points things seem more complicated.

True, we essentially made age always a range, which made the queries consistent. I'm for sticking with that.

**Comment by bussec** [comment]
@bcorrie @schristley We have a potential (long-term) solution for this issue in #551, but this appears to be a larger change to me. Do you think that we can resolve in the near future or should we move to v2.0?

**Comment by schristley** [comment]
I was hoping this could be done for v1.4 as this would make these fields consistent and enhance querying. I think the structure has been agreed upon and implemented in #551. The PR just needs some cleanup to be merged.

**Comment by bcorrie** [comment]
I haven't looked at this in a long time - and I am not too sure about the ramifications of this. It seems like something that requires more deliberation to me and hasn't been on the radar for me, so I would lean towards moving this to v2.0
